src/steps/applications.ts

```diff
@@ -0,0 +1,213 @@
+            appSettings.identityProviderArn,
+          );
+          if (awsAccountIdMatch) {
+            assignData['awsAccountId'] = awsAccountIdMatch[1];
+            assignData['appAccountId'] = awsAccountIdMatch[1];
+          }
+        }
+
+        assignData['awsIdentityProviderArn'] = appSettings.identityProviderArn;
+        assignData['awsEnvironmentType'] = appSettings.awsEnvironmentType;
+        assignData['awsGroupFilter'] = appSettings.groupFilter;
+        assignData['awsRoleValuePattern'] = appSettings.roleValuePattern;
+        assignData['awsJoinAllRoles'] = appSettings.joinAllRoles;
+        assignData['awsSessionDuration'] = appSettings.sessionDuration;
+      } else if (appSettings.githubOrg) {
+        assignData['githubOrg'] = appSettings.githubOrg;
+        assignData['appAccountId'] = appSettings.githubOrg;
+      } else if (appSettings.domain) {
+        // Google Cloud Platform and G Suite apps use `domain` as the account identifier
+        assignData['appDomain'] = appSettings.domain;
+        assignData['appAccountId'] = appSettings.domain;
+      }
+    }
+
+    const appEntity = await jobState.addEntity(
+      createIntegrationEntity({
+        entityData: {
+          source: app,
+          assign: { ...assignData },
+        },
+      }),
+    );
+
+    await jobState.addRelationship(
+      createDirectRelationship({
+        _class: RelationshipClass.HAS,
+        from: accountEntity,
+        to: appEntity,
+      }),
+    );
+
+    //get the groupIds that are assigned this app
+    const groupIds = await apiClient.getGroupsForApp(app.id);
+    for (const groupId of groupIds || []) {
+      const groupEntity = await jobState.findEntity(groupId);
+
+      if (!groupEntity) {
+        throw new IntegrationMissingKeyError(
+          `Expected group with key to exist (key=${groupId})`,
+        );
+      }
+
+      await jobState.addRelationship(
+        createDirectRelationship({
+          _class: RelationshipClass.ASSIGNED,
+          from: groupEntity,
+          to: appEntity,
+        }),
+      );
+    }
+
+    //get the userIds that are assigned to this app individually
+    const userIds = await apiClient.getUsersForApp(app.id);
+    for (const userId of userIds || []) {
+      const userEntity = await jobState.findEntity(userId);
+
+      if (!userEntity) {
+        throw new IntegrationMissingKeyError(
+          `Expected user with key to exist (key=${userId})`,
+        );
+      }
+
+      await jobState.addRelationship(
+        createDirectRelationship({
+          _class: RelationshipClass.ASSIGNED,
+          from: userEntity,
+          to: appEntity,
+        }),
+      );
+    }
+  });
+}
+export const applicationSteps: IntegrationStep<IntegrationConfig>[] = [
+  {
+    id: 'fetch-applications',
+    name: 'Fetch Applications',
+    entities: [
+      {
+        resourceName: 'Okta Application',
+        _type: APPLICATION_ENTITY_TYPE,
+        _class: 'Application',
+      },
```

Comment:
It would be good in time to move to using the pattern of storing these structures in a `constants.ts` file and references them here such as `entities.APPLICATION`. We've been moving other integrations this direction, [as seen in Azure](https://github.com/JupiterOne/graph-azure/blob/06221221a4baa834086a613eb06df1c4c6cd18b2/src/steps/resource-manager/security/index.ts#L191).

Reply:
I like this idea. I wondered, with all these constants scattered all over the app, how useful they would really be for a name change in the future. So, that's a great way to consolidate and make it reasonable to quickly rename things.